build.sbt

```diff
@@ -1,31 +1,62 @@
+ThisBuild / scalaVersion := "3.0.1-RC2"
 
-// also used as a `groupId` by Sonatype
-organization := "ch.epfl.scala"
+lazy val library = project
+  .in(file("library"))
+  .settings(
+    name := "library-example",
+    organization := "ch.epfl.scala", // also used as a `groupId` by Sonatype
+    description := "A library that does nothing useful",
+    crossScalaVersions := Seq("3.0.0", "2.13.6", "2.12.14"),
+    scalaVersion := crossScalaVersions.value.head,
+    libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test,
+    testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+  ).settings(publishSettings)
 
-libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val documentation = project
+  .in(file("site"))
+  .settings(siteSettings)
+  .enablePlugins(MdocPlugin)
+  .dependsOn(library)
 
-import xerial.sbt.Sonatype._
-sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch"))
-// indicate the open source licenses that apply to our project
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-// publish to the Sonatype repository
-publishTo := sonatypePublishToBundle.value
+lazy val siteSettings = {
+  /*
+  Combine scaladoc and mdoc.
+  To do so,
+   1. copy site to target/processed_site
+   2. use mdoc to process site/docs/ **.md
+   3. mdoc outputs new md files in target/processed_site/docs
+   4. scaladoc runs and outputs files in site-ouput
+  */
+  val tmpSiteDir = "processed_site"
+  val outputSiteDir = "site-output"
 
-// documentation website
-enablePlugins(ParadoxPlugin, ParadoxSitePlugin, MdocPlugin, SiteScaladocPlugin)
-mdocIn := sourceDirectory.value / "documentation"
-mdocExtraArguments += "--no-link-hygiene"
-Compile / paradox / sourceDirectory := mdocOut.value
-makeSite := makeSite.dependsOn(mdoc.toTask("")).value
-SiteScaladoc / siteSubdirName := "api"
-paradoxProperties += ("scaladoc.base_url" -> "api")
+  val copySiteTask = Def.task {
+    IO.copyDirectory(
+      baseDirectory.value / "site",
+      target.value / tmpSiteDir
+    )
+  }
 
-// binary compatibility check
-mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+  Def.settings(
+    library / Compile / doc / target := (LocalRootProject / target).value / outputSiteDir,
```

Comment:
Why use `LocalRootProject` instead of the “current” project?

Reply:
so that the site is in `/target` rather than `documentation/target`

Reply:
Yes but why? I think it’s better to not mix `target/` directories across modules.

Reply:
Because given that scaladoc is picked from library, the output files will be written to `library/target` and not to `target` or `site/target` which would make more sense.
I can change it to `site/target` but since it is the documentation of the entire project and not only library or site, I thought it made sense to have it in `target`